lib/directory-tree.js

```diff
@@ -3,63 +3,91 @@
 const FS = require('fs');
 const PATH = require('path');
 const constants = {
-	DIRECTORY: 'directory',
-	FILE: 'file'
-}
+    DIRECTORY: 'directory',
+    FILE: 'file',
+  };
+
+function safeReadDirSync(path) {
+  let dirData = {};
+  try {
+    dirData = FS.readdirSync(path);
+  } catch (ex) {
+    if (ex.code == 'EACCES')
+    //User does not have permissions, ignore directory
+        return null;
+    else throw ex;
+  }
 
-function safeReadDirSync (path) {
-	let dirData = {};
-	try {
-		dirData = FS.readdirSync(path);
-	} catch(ex) {
-		if (ex.code == "EACCES")
-			//User does not have permissions, ignore directory
-			return null;
-		else throw ex;
-	}
-	return dirData;
+  return dirData;
 }
 
-function directoryTree (path, options, onEachFile) {
-	const name = PATH.basename(path);
-	const item = { path, name };
-	let stats;
+function directoryTree(path, options, onEachFile) {
+  const name = PATH.basename(path);
+  const item = { path, name };
+  let stats;
+
+  if (!options) {
+    options = {};
+  }
+
+  try {
+    stats = FS.statSync(path);
+  }
+  catch (e) {
+    return null;
+  }
+
+  if (options.depth == undefined || options.depth == null || options.depth >= 0) {
+    // Skip if it matches the exclude regex
+    if (options && options.exclude && options.exclude.test(path))
+        return null;
+    if (stats.isFile()) {
+      const ext = PATH.extname(path).toLowerCase();
 
-	try { stats = FS.statSync(path); }
-	catch (e) { return null; }
+      // Skip if it does not match the extension regex
+      if (options && options.extensions && !options.extensions.test(ext))
+          return null;
 
-	// Skip if it matches the exclude regex
-	if (options && options.exclude && options.exclude.test(path))
-		return null;  
+      item.size = stats.size;  // File size in bytes
+      item.extension = ext;
+      item.type = constants.FILE;
+      if (onEachFile) {
+        onEachFile(item, PATH);
+      }
+    } else if (stats.isDirectory()) {
+      item.type = constants.DIRECTORY;
+      let childOptions = Object.assign({}, options);
+      if (isNaN(childOptions.depth) == false) {
+        childOptions.depth--;
+      }
+
+      item.children = getChildrenByPath(path, childOptions, onEachFile);
+      if (item.children === null) return null;
```

Comment:
Does this mean that an empty directory will not be included in the result? Should add a test for this.